snuba/migrations/operations.py

```diff
@@ -0,0 +1,50 @@
+from abc import ABC, abstractmethod
+
+from snuba.clusters.cluster import ClickhouseClientSettings, get_cluster
+from snuba.clusters.storage_sets import StorageSetKey
+
+
+class Operation(ABC):
+    """
+    Executed on all the nodes of the cluster.
+    """
+
+    @abstractmethod
+    def execute(self) -> None:
+        raise NotImplementedError
+
+
+class SqlOperation(Operation, ABC):
+    def __init__(self, storage_set: StorageSetKey):
+        self.storage_set = storage_set
```

Comment:
`_storage_set` if you want to make it available to the subclasses but not externally.

Reply:
Updated